src/rgw/store/dbstore/common/dbstore.h

```diff
@@ -63,6 +63,9 @@ struct DBOpObjectInfo {
   std::string tag;
   uint16_t flags;
   uint64_t versioned_epoch;
+  std::list<std::string> versions; /* Queue to store various versions of the objects so
+                               * far created in the head_object.
+                               */
```

Comment:
yes. Only 'olh' will use 'versions' and yes its better if it is avoided.

The main reason to use <versions> list is to find the next latest version in case  of object deletion. I think <row-id> cannot be relied upon. Other alternative is to use mtime but I wasn't sure if it is safe to assume that any versioned object's mtime doesn't change over time (for eg., if transitioned or if only metadata is changed - though I dont know if any such case arise)

Could you please confirm if same mtime persists through out the lifetime of any object version. We can then order the object versions based on mtime.

Reply:
Even if mtime can change, maybe we can store & query based on ctime.

Reply:
@cbodley @mattbenjamin @dang 

To address this and avoid creation of any new OLH object I added version-number to order the version entries and identify the last one to promote/demote as CURRENT. Please review the changes.